lnrpc/routerrpc/router_backend.go

```diff
@@ -1348,24 +1336,26 @@ func MarshalFeatures(feats *lnwire.FeatureVector) []lnrpc.FeatureBit {
 // UnmarshalFeatures converts a list of uint32's into a valid feature vector.
 // This method checks that feature bit pairs aren't assigned together, and
 // validates transitive dependencies.
-func UnmarshalFeatures(
-	rpcFeatures []lnrpc.FeatureBit) (*lnwire.FeatureVector, error) {
-
+func UnmarshalFeatures(rpcFeatures []lnrpc.FeatureBit) *lnwire.FeatureVector {
 	// If no destination features are specified we'll return nil to signal
 	// that the router should try to use the graph as a fallback.
 	if rpcFeatures == nil {
-		return nil, nil
+		return nil
 	}
 
 	raw := lnwire.NewRawFeatureVector()
 	for _, bit := range rpcFeatures {
-		err := raw.SafeSet(lnwire.FeatureBit(bit))
-		if err != nil {
-			return nil, err
-		}
+		// Even though the spec says that the writer of a feature vector
+		// should never set both the required and optional bits of a
+		// feature, it also says that if we receive a vector with both
+		// bits set, then we should just treat the feature as required.
+		// Therefore, we don't use SafeSet here when parsing a peer's
```

Comment:
yeah the spec is confusing - but based on the description, when we are setting the bits we should always make sure not to set both, and only when we are receiving the bits we can then treat the feature as mandatory? so we should still have this check when we are `parseQueryRoutesRequest` or `extractIntentFromSendRequest` since we are creating the bits?

Reply:
but we are not creating the bits in `parseQueryRoutesRequest` and `extractIntentFromSendRequest`. For those we are reading bits we got from the peer in an invoice. When we set our own bits, we do only set one.

Reply:
I think for sending payments that's only true when it's using `PaymentRequest` aka the invoice, as the caller can set `DestFeatures`.

Reply:
sure but i think it's ok for us to be lenient for any externally created feature vector. ie, including those set in DestFeatures - as perhaps they were extracted from a payReq by the caller. 

Ie, i think we only need to be strict with the ones that we ourselves are generating from within LND and including in invoices/feature vectors we create. 

Thoughts?